tools/src/main/scala/scala/scalanative/build/Platform.scala

```diff
@@ -41,4 +41,21 @@ object Platform {
    *    true if `FreeBSD`, false otherwise
    */
   lazy val isFreeBSD: Boolean = osUsed.contains("freebsd")
+
+  /** Test for the target type
+   *
+   *  @return
+   *    true if `msys`, false otherwise
+   */
+  lazy val isMsys: Boolean = target.endsWith("msys")
+
+  /** Test for the target type
+   *
+   *  @return
+   *    true if `cygnus`, false otherwise
+   */
+  lazy val isCygwin: Boolean = target.endsWith("cygnus")
+
+  private lazy val target =
+    System.getProperty("target.triple", "unknown").toLowerCase(Locale.ROOT)
```

Comment:
That's a good idea, but `target.triple` is currently never set in system properties. 
We might use `java.lang.System.setProperty("target.triple", ???)` somewhere early in the build, but I'm not sure what would be the best place to put it. Probably in the `scala.scalanative.build.NativeConfig.Impl` constructor, because we would need to propagate this information on every update to NativeConfig. 
Because that `nativeConfig` might change, eg. when setting it manually in sbt, the target and the values depending on it should not by `lazy val` but `def`

Reply:
What about something like this (in scala.scalanative.build.NativeConfig.Impl line 255)?
```scala
    def withTargetTriple(value: Option[String]): NativeConfig =
      copy(targetTriple = value)
      System.setProperty("target.triple", value.getOrElse(""))
```

Reply:
I can set the target.triple like this:

```bash
sbt 'set nativeConfig ~= {_.withTargetTriple("x86_64-pc-linux-gnu")}; show nativeConfig'
```

but the value reported by `clang` would be a good default value.

Reply:
had to remove the default `targetTriple`, it was causing test failures that I was unable to diagnose.